server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -532,10 +532,9 @@ private Tuple<RemoteStoreRefreshListener, RemoteStorePressureService> mockIndexS
             new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS),
             threadPool
         );
-        RemoteStorePressureService remoteStorePressureService = new RemoteStorePressureService(clusterService, Settings.EMPTY);
+        RemoteStorePressureService remoteStorePressureService = indexShard.getRemoteStorePressureService();
         when(shard.indexSettings()).thenReturn(indexShard.indexSettings());
         when(shard.shardId()).thenReturn(indexShard.shardId());
-        remoteStorePressureService.afterIndexShardCreated(shard);
```

Comment:
Removing this looks right as the event would already be handled by the listener. But line 470 above shows that `shard` is a mock. Should this be removed in that case?

Reply:
Mock `shard` is mostly getting all the details from `indexShard`. `remoteStorePressureService.afterIndexShardCreated` is called once indexShard is created.